Source/ACE/Entity/Generator.cs

```diff
@@ -0,0 +1,100 @@
+﻿using ACE.Entity.Enum;
+using ACE.Network.Enum;
+using ACE.Network.Motion;
+using System;
+using System.Diagnostics;
+
+namespace ACE.Entity
+{
+    public class Generator : WorldObject
+    {
+        public Generator(ObjectGuid guid, AceObject baseAceObject)
+            : base((ObjectType)baseAceObject.ItemType, guid)
+        {
+            Name = baseAceObject.Name ?? "NULL";
```

Comment:
why all the commented code in a new class?  Can you document why all this isn't necessary and just nuke it?